### ISSUE TITLE
AndroidLifecycleScopeProvider do not work

### ISSUE THREAD

**Issue by jaychang0917** [uber/AutoDispose#93]
I tried the `AndroidLifecycleScopeProvider` implementation with the following code, but it doesn't work and keep emitting when onStop()` is called.

```kotlin
override fun onStart() {
    super.onStart()
    Observable.interval(0, 1, TimeUnit.SECONDS)
      .doOnDispose { println("dispose") }
      .to(ObservableScoper(AndroidLifecycleScopeProvider.from(this)))
      .subscribe {
        println("emit:" + it)
      }
  }

  override fun onStop() {
    super.onStop()
    println("onStop")
  }
```

Thanks.

**Comment by hzsweers** [comment]
Interesting, I'll give it a shot with those conditions and try to repro

**Comment by hzsweers** [comment]
Repro'd, but the issue is that AutoDispose isn't quite acting *wrong* here. I'll need to ask for clarification, but it appears that `getCurrentState()` reports the previous state if you call it from within one of the `on_____` lifecycle callbacks.

<img width="782" alt="screen shot 2017-09-29 at 1 44 46 am" src="https://user-images.githubusercontent.com/1361086/31008172-f67fc3bc-a4b7-11e7-80aa-310c664fbaa3.png">

**Comment by hzsweers** [comment]
To be clear - I want to make sure this is expected behavior. Assuming it is, the fix would be to just adjust the backfill offsets by 1

**Comment by jaychang0917** [comment]
According to [this](https://developer.android.com/topic/libraries/architecture/lifecycle.html#lc), seems you are correct.
![](https://developer.android.com/images/topic/libraries/architecture/lifecycle-states.png)

**Comment by hzsweers** [comment]
Ah cool, thanks for that. Basically we need to start with "initialized". Interested in sending a PR? Otherwise I'll do it tomorrow.

**Comment by jaychang0917** [comment]
Waiting for your fix👍

**Comment by jaychang0917** [comment]
Will `android-archcomponents` be released in 0.2.x branch?

**Comment by hzsweers** [comment]
It would be part of 0.3.0
